// @turnkey/sdk-browser/dist/turnkey-helpers.js
'use strict';

// ----------------------------
// CURVE_SECP256K1 Accounts
// ----------------------------
// Ethereum
const defaultEthereumAccountAtIndex = (pathIndex) => {
    return {
        curve: "CURVE_SECP256K1",
        pathFormat: "PATH_FORMAT_BIP32",
        path: `m/44'/60'/${pathIndex}'/0/0`,
        addressFormat: "ADDRESS_FORMAT_ETHEREUM",
    };
};
const DEFAULT_ETHEREUM_ACCOUNTS = [
    defaultEthereumAccountAtIndex(0),
];
// ----------------------------
// CURVE_ED25519 Accounts
// ----------------------------
// Solana
const defaultSolanaAccountAtIndex = (pathIndex) => {
    return {
        curve: "CURVE_ED25519",
        pathFormat: "PATH_FORMAT_BIP32",
        path: `m/44'/501'/${pathIndex}'/0'`,
        addressFormat: "ADDRESS_FORMAT_SOLANA",
    };
};
const DEFAULT_SOLANA_ACCOUNTS = [
    defaultSolanaAccountAtIndex(0),
];

exports.DEFAULT_ETHEREUM_ACCOUNTS = DEFAULT_ETHEREUM_ACCOUNTS;
exports.DEFAULT_SOLANA_ACCOUNTS = DEFAULT_SOLANA_ACCOUNTS;
exports.defaultEthereumAccountAtIndex = defaultEthereumAccountAtIndex;
exports.defaultSolanaAccountAtIndex = defaultSolanaAccountAtIndex;
